File: structures/dlc_list/node.go
package dlclist

type Node[T any] struct {
	Value T
	Front *Node[T]
	Back  *Node[T]
}

func NewNode[T any](value T) *Node[T] {
	return &Node[T]{
		Value: value,
		Front: nil,
		Back:  nil,
	}
}

func (node *Node[T]) GetValue() T {
	return node.Value
}

func (node *Node[T]) SetValue(value T) {
	node.Value = value
}

func (node *Node[T]) GetFront() *Node[T] {
	return node.Front
}

func (node *Node[T]) SetFront(front *Node[T]) {
	node.Front = front
}

func (node *Node[T]) GetBack() *Node[T] {
	return node.Back
}

func (node *Node[T]) SetBack(back *Node[T]) {
	node.Back = back
}

File: dlc_list/node.c
#include "node.h"
#include <stdio.h>
#include <stdlib.h>

Node *newNode(int data)
{
    Node *node = (Node *)malloc(sizeof(Node));
    node->data = data;
    node->front = NULL;
    node->back = NULL;
    return node;
}

Node *getFront(Node *node)
{
    return node->front;
}

Node *getBack(Node *node)
{
    return node->back;
}

void setFront(Node *node, Node *front)
{
    node->front = front;
}

void setBack(Node *node, Node *back)
{
    node->back = back;
}
